src/gsy_e/models/market/grid_fees/constant_grid_fees.py

```diff
@@ -52,7 +52,7 @@ def update_forwarded_bid_trade_original_info(self, trade_original_info, market_b
             return None
         trade_offer_info = TradeBidOfferInfo(
             original_bid_rate=market_bid.original_price / market_bid.energy,
-            propagated_bid_rate=market_bid.energy_rate,
+            propagated_bid_rate=trade_original_info.propagated_bid_rate or market_bid.energy_rate,
```

Comment:
This seems something unrelated to this PR. Can you please explain?

Reply:
Indeed, this is a bug that Luis correctly pointed out to, basically when we match with DOF, `market_bid.energy_rate` is not used, we use the rate from the requirement then (which is reflected in the  propagated_bid_rate field of the TradeBidOfferInfo)